p2p/src/handshake.rs

```diff
@@ -0,0 +1,264 @@
+//! This is a WIP module containing the refactored state-machine based
```

Comment:
I think that this module should be removed. @Erigara fixed `p2p` to look much better, and this is no longer necessary.